org-scan/scan_org.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/sh
+
+CACHEDIR=${HOME}/Documents/caulking-org-scan/cache
+
+list_repos() {
+    for page in 1 2 3 4; do
+    hub api /orgs/cloud-gov/repos\?per_page=100\&page=$page | 
+        #jq -r '.[] | select (.archived == true ) | .name'
+        jq -r '.[] | .name'
+    done
+}
+
+scan() {
+    config=gitleaks.toml
+    if [ -r $repo.toml ]; then
+        config=$repo.toml
+        echo Using config: $config
+    fi
+    cmd='gitleaks --config=./$config --repo-path=$1'
+    eval echo $cmd
+    eval $cmd
+}
+
+cache(){
+    local repo=$1
+    git_url=git@github.com:cloud-gov/$repo.git
+    if [ -d $cache_dir/$repo ] ; then
+
+        echo ... fetching ...
+        (cd $cache_dir/$repo; git fetch --all)
+    else
+        echo ... cloning ...
+        git clone $git_url $cache_dir/$repo
+    fi
+}
+
+# For finding commits to exclude, not currently used
+commits() {
+    cat $1 | grep -v WARN | jq '.commit' | sort | uniq >> gitleaks.toml
+}
+
+make_repo_list() {
+  echo "Making repo list"
+  list_repos | sort | 
+    grep -v openbrokerapi |
+    grep -v stratos |
+    grep -v 'cg-release' | # never used by us
+    grep -v 'cf-example-suitecrm' | # we haven't worked this in 4 years
```

Comment:
is the extra logic to ignore these, especially the ones that just don't move, actually worth it?

in other words - it's possible we could commit, or could have committed, secrets to our fork of stratos. What do we save by not checking it?

Reply:
These exceptions do _not_ apply to real-time pre-commit hooks. I don't scan these repositories retrospectively because they're HUGE (and take about 30m to scan) and they're not ours (or are relics), so shouldn't be changing.